### DISCUSSION TITLE
Multilevel Recursive Bind Support with minimal bind names.

### DISCUSSION THREAD

**Discussion by jrmycanady** [aarondl/sqlboiler#933] [General]
I am looking to validate that it's not possible to have Bind recurse more than one level without defining unique bind names. What I mean by that is I would like to have structs containing other structs and utilize the Bind method to bind data to all structs via unique column names. This is possible as bind does recurse, but it does not appear to work with sub structs without large unique names.

That doesn't sound very clear so here is a structure I am looking to dump into.

```go
// All item models are the same as:
type Model struct {
	Name string `boil:"name,bind"`
}

type ItemAResponse struct {
	model.ItemA `boil:,bind"`
	ItemB model.ItemBResponse `boil:itemb,bind`
}
type ItemBResponse struct {
	model.ItemB `boil:,bind`
	ItemC1 ItemCResponse `boil:itemc1,bind`
	ItemC2 ItemCResponse `boild:itemc2,bind`
}
type ItemCResponse struct {
	model.ItemC `boil:",bind"`
}
```

The SQL column names would be as follows.
```
name -> Would be ItemAResponse.Name
itemb.name -> Would be ItemAReponse.ItemB.Name
itemb.itemc1.name -> Would be ItemAResponse.ItemB.ItemC1.name
itemb.itemc2.name -> Would be ItemAResponse.ItemB.ItemC2.name
```

Essentially bind would be aware of the depth of the prefix and only apply columns that match that depth.

The use case, as wonky as it may be, is to allow queries to span multiple levels of joins. I have a working query solution that can build the queries (select,where,join) easily enough so all column names are defined and used correctly. But I never looked to see if bind could handle such a situation. So far in testing it appears it cannot until I manually define the full column name in any sub structs.

**Comment by aarondl** [comment]
There shouldn't be any problem with this. Can you produce a test or a reproduce where bind is failing to properly do this? It'd be a bug we should correct.

**Reply by jrmycanady** [reply]
Thanks for the reply. I will see about making time to build a test to reproduce this week.
